File: Front End/src/app/category/category.ts
import { INursery } from "../nursary/nursery";
import { ISaplingType } from "../sapling-type/sapling-type";

export interface ICategory {
    id: string;
    name: string;
    isActive: boolean;
    saplingTypes: ISaplingType[];
    nurseries: INursery[];
}

File: Back End/Application.Domain/Models/Category.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.Domain.Models
{
    // كلاس الصنف 
    public class Category
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
        public bool IsActive { get; set; } = true;

        public List<SaplingType> SaplingTypes { get; set; } = new List<SaplingType>();
        public List<Nursery> Nurseries { get; set; } = new List<Nursery>();

        public Category()
        {
            Id = Guid.NewGuid();
        }
    }
}
